sagemaker_inference/files/task.py

```diff
@@ -0,0 +1,159 @@
+import tempfile
+import pickle
+import json
+import csv
+import logging
+from argparse import ArgumentParser
+import datetime
+import sys
+from flask import Flask, request
+import flask
+import csv
+from grants_tagger.predict import predict_tags
+
+csv.field_size_limit(sys.maxsize)
+
+
+def yield_grants(input_file):
+    """yield grants from s3 location line by line"""
+    with open(input_file, "r") as tf:
+        csv_reader = csv.DictReader(tf, delimiter=",", quotechar='"')
+        for line in csv_reader:
+            yield line
+
+
+app = Flask(__name__)
+
+
+@app.route("/ping", methods=["GET"])
+def ping():
+    # Load model
+    model = load_model("/opt/model", parameters=None)
+    # Check if the classifier was loaded correctly
+    health = model is not None
+    status = 200 if health else 404
+    return flask.Response(response="\n", status=status, mimetype="application/json")
+
+
+@app.route("/invocations", methods=["POST"])
+def tag_grants():
+    data = request.get_data().decode("utf-8")
+    input_csv = csv.DictReader(data.splitlines())
+    TG = TagGrants()
+    TG(input_file=input_csv, output_file="output.csv")
+    return flask.Response(response=TG.output_file, status=200)
+
+
+class TagGrants:
+    """
+    Task to read grants from file, tag and output
+        to CSV file
+    Args:
+        input_file: Grants data source location
+        output_file: Location to output data to
+        model_path: The path to the model file
+        label_binarizer_path: The path the binarizer pkl file
+        cap: (default=None) a cap of number of records to process (testing)
+    """
+
+    def __init__(
+        self,
+        input_file,
+        output_file,
+        approach,
+        threshold,
+        model_path,
+        label_binarizer_path,
+        cap,
+    ):
+        super(TagGrants, self).__init__()
```

Comment:
I'm not sure `super` is doing anything here as this class is inheriting from `object`. Unless I have missed something @nsorros ?

Reply:
maybe copilot wrote it :) agree, i dont think its needed

Reply:
I took it over from some code Jeff wrote when task.py was run on the engineering side. But if it's not necessary it can certainly go